tests/contrib/scrapy/test_pipeline_cerberus.py

```diff
@@ -0,0 +1,147 @@
+            TestItem(),
+            {SETTING_CERBERUS: {TestItem: cerberus_test_schema}},
+            [assert_in_cerberus_stats(TestItem), STATS_ITEM_ERRORS],
+            id="validator is {} type, loads from a python dict".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: [f"{os.getcwd()}/contrib/scrapy/schema.json"]},
+            [assert_in_cerberus_stats(Item), "{}".format(STATS_ITEM_ERRORS)],
+            id="validator is {} type, loads from path to schema".format(Item.__name__),
+        ),
+        pytest.param(
+            TestItem(),
+            {
+                SETTING_CERBERUS: {
+                    TestItem: "tests.fixtures.validators.cerberus_test_schema"
+                }
+            },
+            [assert_in_cerberus_stats(TestItem), "{}".format(STATS_ITEM_ERRORS)],
+            id="validator is {} type, loads from path to a python dict".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: {TestItem: [cerberus_test_schema]}},
+            [assert_in_cerberus_stats(TestItem), STATS_ITEM_ERRORS],
+            id="validator is {} type, validators are in a list repr".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: [cerberus_test_schema]},
+            ["spidermon/validation/fields/errors/missing_required_field"],
+            id="Missing required fields",
+        ),
+    ],
+)
+def test_stats_in_pipeline(item, settings, cases):
+    crawler = get_crawler(settings_dict=settings)
+    pipe = ItemValidationPipeline.from_crawler(crawler)
+    pipe.process_item(item, None)
+    for case in cases:
+        casechecker = lambda x: x in pipe.stats.stats.get_stats()
+        assert casechecker(case)
+
+
+@pytest.mark.parametrize(
+    "item,settings,cases",
+    [
+        pytest.param(
+            TestItem({"url": "example.com"}),
+            {SETTING_CERBERUS: [cerberus_test_schema]},
+            [STATS_ITEM_ERRORS],
+            id="processing simple items without errors",
+        ),
+        pytest.param(
+            TreeItem(
+                {
+                    "quotes": {"author": "Vipul Gupta", "quote": "Life is vanilla"},
+                    "child": "https://example.com",
+                }
+            ),
+            {SETTING_CERBERUS: [cerberus_tree_schema]},
+            [STATS_ITEM_ERRORS],
+            id="processing nested items without errors",
+        ),
+    ],
+)
+def test_stats_not_in_pipeline(item, settings, cases):
+    crawler = get_crawler(settings_dict=settings)
+    pipe = ItemValidationPipeline.from_crawler(crawler)
+    pipe.process_item(item, None)
+    for case in cases:
+        casechecker = lambda x: x not in pipe.stats.stats.get_stats()
+        assert casechecker(case)
```

Comment:
why not `assert case not in pipe.stats.stats.get_stats()`?

Reply:
Dang it, I should make a habit of re-reading my own code, to idiot-proof it.